agent/stats/service_connect_linux.go

```diff
@@ -63,29 +68,203 @@ func (sc *ServiceConnectStats) retrieveServiceConnectStats(task *apitask.Task) {
+				metricCounts = append(metricCounts, &metricCountForCountersAndGauges)
+			case prometheus.MetricType_GAUGE:
+				metricValues = append(metricValues, metric.Gauge.Value)
+				// MetricCount for Gauge will always be [1]
+				metricCounts = append(metricCounts, &metricCountForCountersAndGauges)
+			case prometheus.MetricType_HISTOGRAM:
+				for _, bucket := range metric.Histogram.Bucket {
+					// We do not want to add the metricValue if it is +Inf
+					if math.IsInf(bucket.GetUpperBound(), 0) {
+						continue
+					}
+					metricValues = append(metricValues, bucket.UpperBound)
+
+					// Prometheus histogram CumulativeCount is type *uint64, TACS wants *int64.
+					var metricCount int64
+					if bucket.GetCumulativeCount() <= uint64(math.MaxInt64) {
+						metricCount = int64(bucket.GetCumulativeCount())
+					} else {
+						metricCount = math.MaxInt64
+						logger.Warn("Service Connect histogram metric is emitting a value larger than max int64", logger.Fields{
+							field.TaskID:            taskId,
+							"metric":                v.Type.String(),
+							"bucketCumulativeCount": bucket.GetCumulativeCount(),
+						})
+					}
+					metricCounts = append(metricCounts, &metricCount)
+				}
+				metricCounts = convertHistogramMetricCounts(metricCounts)
+			default:
+				logger.Warn("Service connect stats received invalid Metric type", logger.Fields{
+					field.TaskID: taskId,
+					"metric":     v.Type.String(),
+				})
+				continue
+			}
+
+			generalMetric := &ecstcs.GeneralMetric{}
+			generalMetric.MetricName = v.Name
+			generalMetric.MetricValues = metricValues
+			generalMetric.MetricCounts = metricCounts
+
+			// Get metric dimensions
+			var dimensions []*ecstcs.Dimension
+			if metric.Label != nil {
+				for _, d := range metric.Label {
+					// TODO [SC]: New stats endpoint will surface a new dimension which indicates what is the MetricType.
+					dimension := &ecstcs.Dimension{
+						Key:   d.Name,
+						Value: d.Value,
+					}
+					dimensions = append(dimensions, dimension)
+				}
+			}
+
+			dimensionAsString := sortAndConvertDimensionsintoStrings(dimensions)
+
+			if generalMetricsWrapper, ok := statsMap[dimensionAsString]; !ok {
+				// Dimension does not exist in statsMap, add it to the statsMap
+				generalMetricsList := []*ecstcs.GeneralMetric{generalMetric}
+				generalMetricsWrapper = &ecstcs.GeneralMetricsWrapper{
+					Dimensions:     dimensions,
+					GeneralMetrics: generalMetricsList,
+				}
+				statsMap[dimensionAsString] = generalMetricsWrapper
+			} else {
+				// Add this metric to the metric list for the already existing dimesion
+				generalMetricsWrapper.GeneralMetrics = append(generalMetricsWrapper.GeneralMetrics, generalMetric)
+			}
+		}
+	}
+
+	statsCollectedList := []*ecstcs.GeneralMetricsWrapper{}
+	for _, gm := range statsMap {
+		statsCollectedList = append(statsCollectedList, gm)
+	}
+
+	sc.resetStats()
+	sc.setStats(statsCollectedList)
+	sc.SetStatsSent(false)
```

Comment:
nit(not blocking): Should this chunk be moved back to `retrieveServiceConnectStats()`? since `parseStats` should only be meant to parse the stats as name suggests

Reply:
Made the change.